drivers: video: Add Omnivision OV7675 driver into the existing OV7670 driver.

mjs513 opened the pull request:
This PR is a clean PR of [video: ov7675: add Omnivision OV7675 sensor support and basic controls](https://github.com/zephyrproject-rtos/zephyr/compare/main...mjs513:zephyr:Add-New-OV7675-Driver) PR per your request.

avolmat-st commented on drivers/video/ov7670.c:
```diff
@@ -4,33 +4,22 @@
-#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(pwdn_gpios)
-	struct gpio_dt_spec pwdn;
-#endif
+	const struct gpio_dt_spec reset;
```
It should still be possible to keep reset & pwdn conditionnal by switching to usage of DT_ANY_COMPAT_HAS_PROP_STATUS_OKAY Instead.

Aka something like:

#if DT_ANY_COMPAT_HAS_PROP_STATUS_OKAY(ovti,ov7670, reset)
   const ...
#endif

and when adding the 2nd sensor support, simply use

#if DT_ANY_COMPAT_HAS_PROP_STATUS_OKAY(ovti,ov7670, reset) || 
     DT_ANY_COMPAT_HAS_PROP_STATUS_OKAY(ovti,ov7675, reset)
     ...

avolmat-st commented on dts/bindings/video/ovti,ov7675.yaml:
```diff
@@ -0,0 +1,24 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: OV7670 CMOS video sensor
```
OV7670 -> OV7675

avolmat-st commented on dts/bindings/video/ovti,ov7675.yaml:
```diff
@@ -0,0 +1,24 @@
+# Copyright 2024 NXP
```
Copyright NXP ?

mjs513 commented on drivers/video/ov7670.c:
```diff
@@ -616,27 +564,28 @@ static DEVICE_API(video, ov7670_api) = {
-	VIDEO_DEVICE_DEFINE(ov7670_##inst, DEVICE_DT_INST_GET(inst), NULL);
-
-DT_INST_FOREACH_STATUS_OKAY(OV7670_INIT)
+#define OV7670_INIT(n, id)                                \
```
its actually used for specifying the format table:
```
	static const struct ov767x_config ov767x_config_##n = {                                    \
		.bus = I2C_DT_SPEC_INST_GET(n),                                                    \
		.reset = GPIO_DT_SPEC_INST_GET_OR(n, reset_gpios, {}),                             \
		.pwdn = GPIO_DT_SPEC_INST_GET_OR(n, pwdn_gpios, {}),                               \
		.fmts = ov##id##_fmts,            \
	};
```

avolmat-st replied:
Oh yes indeed, didn't see that it was part of .fmts = ... 

However, same as for the previous comment, if you have a system with both ov7670 and ov7675, this will lead to duplicated structures. So you should for example also embed the ##id within the structures name.

mjs513 replied:
Been playing with this for a while and seems to be causing me compile issues.  As a quick test I did:
```

struct ov7670_data {
	struct ov767x_ctrls ctrls;
	struct video_format fmt;
};

struct ov7675_data {
	struct ov767x_ctrls ctrls;
	struct video_format fmt;
};
```
and associated changes:
```
	static struct ov767x_data ov##id##_data##n;                                                 \
                                                                                                   \
	DEVICE_DT_INST_DEFINE(n, ov767x_init, NULL, &ov##id##_data##n, &ov767x_config##n,          \
			      POST_KERNEL, CONFIG_VIDEO_INIT_PRIORITY, &ov767x_api);   
```
gives several errors but think it stems from this one:
```
/home/my_new_zephyr_folder/zephyr/drivers/video/ov7670.c:705:35: error: storage size of 'ov7670_data0' isn't known
  705 |         static struct ov767x_data ov##id##_data##n; 
```

A typical error is where I call it in the driver I get:
```
In function 'ov767x_set_fmt':
/home/my_new_zephyr_folder/zephyr/drivers/video/ov7670.c:463:26: error: invalid use of undefined type 'struct ov767x_data'
  463 |         if (!memcmp(&data->fmt, fmt, sizeof(data->fmt))) {
```

mjs513 commented on drivers/video/ov7670.c:
```diff
@@ -190,37 +123,26 @@ const struct ov7670_resolution_cfg OV7670_RESOLUTION_VGA = {
 	/* configure the output timing */
 	/* PCLK does not toggle during horizontal blank, one PCLK, one pixel */
-	{OV7670_COM10, 0x20}, /* COM10 */
+	{OV7670_COM10, 0x03}, /* COM10 */
```
Darn it - did it again forgot to change it back.

mjs513 commented on drivers/video/ov7670.c:
```diff
@@ -190,37 +123,26 @@ const struct ov7670_resolution_cfg OV7670_RESOLUTION_VGA = {
 
 /* OV7670 definitions */
-#define OV7670_PROD_ID 0x76
+#define OV7670_PROD_ID    0x76
```
Weird - ran it through clang and check patch as well.

mjs513 replied:
hopefully fixed - checkpatch did not pick up errors

mjs513 commented on drivers/video/ov7670.c:
```diff
@@ -397,67 +377,38 @@ static int ov7670_set_fmt(const struct device *dev, struct video_format *fmt)
 
-	LOG_ERR("Unsupported format");
-	return -ENOTSUP;
+	return 0;
```
Think I may disagree on this one.  That return 0 will only be reached writing the registers is successful.  If a resolution is not supported or fails to right its caught with the if statement
```
		if (config->fmts[i].width_min == fmt->width && config->fmts[i].height_min == fmt->height &&
		    config->fmts[i].pixelformat == fmt->pixelformat) {
......
			}
			if (ret < 0) {
				LOG_ERR("Resolution not set or not supported!");
				return ret;
			}
		}
```
If the write is successful `video_write_cci_multiregs8` returns 0 value so I just did a return 0 instead of return ret

avolmat-st replied:
Assume the application sets the format RGB565 1024x768 ok ?

It will loop through each entry of the config table, one after the other until it finds an entry which has the same format / width / height of what is requested by the set_fmt. Obviously it will never hit and this won't enter into the if statement. It will thus loop throw the config table until it reachs the end of the while loop and will then return 0, which is wrong here since this format is not supported.

mjs513 replied:
Understand now and got it fixed.  Testing the changes as I go.

mjs513 commented on drivers/video/ov7670.c:
```diff
@@ -291,9 +316,41 @@ static const struct video_reg8 ov7670_init_regtbl[] = {
 
+
+static const struct video_reg8 ov767x_rgb565_regs[] = {
+	{OV7670_COM7, COM7_RGB_FMT}, /* Selects RGB mode */
```
Will update and make appropriate changes.

mjs513 commented on drivers/video/ov7670.c:
```diff
@@ -122,7 +123,31 @@ struct ov7670_data {
+#define COM15_OUT_00_FF      0xC0 /* Output data range 00 to FF  */
+#define COM15_FMT_RGB_NORMAL 0x00 /* Normal RGB normal output    */
+#define COM15_FMT_RGB565     0x10 /* Normal RGB 565 output       */
+#define COM15_FMT_RGB555     0x30 /* Normal RGB 555 output       */
```
Updating

avolmat-st commented on drivers/video/ov7670.c:
```diff
@@ -346,20 +403,83 @@ static const struct video_format_cap fmts[] = {
 
-static int ov7670_get_caps(const struct device *dev, struct video_caps *caps)
+#if DT_HAS_COMPAT_STATUS_OKAY(ovti_ov7675)
+static const struct video_reg8 ov767x_regs_vga[] = {
```
Yes, however it is will possible that there are both a ov7670 AND a ov7675 available on a system, leading to having the same symbol twice, one from line 3386 and one from line 411.

mjs513 replied:
Note there is a second table, qvga, that will have the same problem.  Fixed and revamped how setFormat sets the resolution.

avolmat-st commented on drivers/video/ov7670.c:
```diff
@@ -4,33 +4,22 @@
-#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(pwdn_gpios)
-	struct gpio_dt_spec pwdn;
-#endif
+	const struct gpio_dt_spec reset;
```
What do you mean by failing when OV7675 is selected ? I don't think what you propose above will generate the same behavior as expected. Could you confirm that it leads to the expected behavior ?

avolmat-st commented:
@mjs513 

Please do not add a commit to fix review comments. You have to update the existing commits in order to fix the issue related to what is in each commit.
Moreover, the code you are providing does not compile and add new structures that are not used within the driver.

Please fix that before requesting again a new review.

mjs513 commented on drivers/video/ov7670.c:
```diff
@@ -33,6 +33,34 @@ struct ov767x_data {
 };
 
+#if DT_HAS_COMPAT_STATUS_OKAY(ovti_ov7670)
+struct ov7670_config {
```
quick answer - yes if you want to use the '`id` to create separate instances to support having both cameras connected at the same time this was one of your suggestions

>However, same as for the previous comment, if you have a system with both ov7670 and ov7675, this will lead to duplicated structures. So you should for example also embed the ##id within the structures name.

avolmat-st commented:
> Thought you would want it as new commit - going to be a pain to do it in the separate commits. In addition other than the 0x20 comment they seemed to all apply to the combined version

Only commits which have been reviewed and are corrects will be merged into the tree. We do not merge series of commits fixing each others within the same PR.

Cf https://docs.zephyrproject.org/latest/contribute/contributor_expectations.html

Especially this section:https://docs.zephyrproject.org/latest/contribute/contributor_expectations.html#pr-requirements, point 3 (Squash Intermediary or Non-Final Development History) and 4 (Ensure Clean History Before Submission)

avolmat-st commented:
Hi @mjs513,

I will have a look at the code itself. I don't know how to make comment on the commit log itself so I write it here.
This is how it looks like now:

```
commit a1a7b03c47f93c32a6a947523fe5a017c377c430 (HEAD -> Add-New-OV7675-Driver)
Author: Mike S <CyberMerln@gmail.com>
Date:   Sat Sep 20 10:13:23 2025 -0400

    drivers: video: Commit adds the OV7675 driver to the existing OV7670 driver
    
    drivers: video: Commit adds the OV7675 driver to the existing OV7670 driver
    
    Updates ov7670.c to incorporate the OV7675 driver into the OV7670 driver.
    
    Signed-off-by: Michael Smorto <CyberMerln@gmail.com>

commit 4a441e429ef5164938660da1587d391791d1d00a
Author: Mike S <CyberMerln@gmail.com>
Date:   Sat Sep 20 10:04:29 2025 -0400

    drivers: video: ov7670 driver changes in prep for adding OV7675
    
    drivers: video: ov7670 driver changes in prep for adding OV7675
    
    Updates ov7670.c in prep for incorporation of OV7675. See previous PR.
    
    Signed-off-by: Michael Smorto <CyberMerln@gmail.com>

```

Aka, the short message like (first line) is repeated again within the commit log. Could you fix that ?
Moreover, only commits remains, so it is not possible to have **See previous PR** in a commit log. You need to describe what the commit does.

mjs513 commented:
@avolmat-st  - @KurtE 

All changes incorporated but can not seem to update the com10 register in prep commit (drivers: video: ov7670 driver changes in prep for adding OV7675)  but it is fixed in the full commit.

I did manage to fix dual camera mode so should be ready to try when PRs are incorporated for multiple streams.

Tested on the GIGA - you are getting compile errors let me know what you are seeing.

avolmat-st commented on drivers/video/ov7670.c:
```diff
@@ -4,33 +4,22 @@
-#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(pwdn_gpios)
-	struct gpio_dt_spec pwdn;
-#endif
+	const struct gpio_dt_spec reset;
```
@mjs513 
Sorry, I am pretty sure it is changed in the first commit. Cf below the beginning of the commit:

```diff
commit 4a441e429ef5164938660da1587d391791d1d00a
Author: Mike S <CyberMerln@gmail.com>
Date:   Sat Sep 20 10:04:29 2025 -0400

    drivers: video: ov7670 driver changes in prep for adding OV7675
    
    drivers: video: ov7670 driver changes in prep for adding OV7675
    
    Updates ov7670.c in prep for incorporation of OV7675. See previous PR.
    
    Signed-off-by: Michael Smorto <CyberMerln@gmail.com>

diff --git a/drivers/video/ov7670.c b/drivers/video/ov7670.c
index 448b0e659bc..8afbb90ae6f 100644
--- a/drivers/video/ov7670.c
+++ b/drivers/video/ov7670.c
@@ -4,33 +4,22 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#define DT_DRV_COMPAT ovti_ov7670
-
 #include <zephyr/drivers/gpio.h>
 #include <zephyr/drivers/i2c.h>
 #include <zephyr/drivers/video.h>
 #include <zephyr/drivers/video-controls.h>
 #include <zephyr/logging/log.h>
 
+#include "video_common.h"
 #include "video_ctrls.h"
 #include "video_device.h"
 
 LOG_MODULE_REGISTER(video_ov7670, CONFIG_VIDEO_LOG_LEVEL);
 
-/* Initialization register structure */
-struct ov7670_reg {
-       uint8_t reg;
-       uint8_t cmd;
-};
-
 struct ov7670_config {
        struct i2c_dt_spec bus;
-#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(reset_gpios)
-       struct gpio_dt_spec reset;
-#endif
-#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(pwdn_gpios)
-       struct gpio_dt_spec pwdn;
-#endif
+       const struct gpio_dt_spec reset;
+       const struct gpio_dt_spec pwdn;
 };
 
 struct ov7670_ctrls {
@@ -43,63 +32,7 @@ struct ov7670_data {
        struct video_format fmt;
 };
```

mjs513 commented:
> I can't help with Github desktop, never used that.
Playing with just using command line - any hints

avolmat-st commented on drivers/video/ov7670.c:
```diff
@@ -395,69 +378,53 @@ static int ov7670_set_fmt(const struct device *dev, struct video_format *fmt)
 
 	memcpy(&data->fmt, fmt, sizeof(data->fmt));
 
+	uint32_t a, b, c;
```
Usage of variable doesn't help in understanding the if statement. I think it is better to keep as before, that done it is very easy to understand what is checked.

mjs513 replied:
Only reason changed was that if I leave the I get a error message from twister that the line is too long - I can put it back no issue if that is what you prefer.  Just let me know.  Will leave this open for now

avolmat-st replied:
In such case it is possible to cut the line as below are something similar ..
```
  if (fmt.width == xx.width &&
      fmt.height == xx.height &&
      fmt.pixelformat == xx.pixelformat) {
```

avolmat-st commented on drivers/video/ov7670.c:
```diff
@@ -395,69 +378,53 @@ static int ov7670_set_fmt(const struct device *dev, struct video_format *fmt)
+	}
+
+	if (ret < 0) {
+		LOG_ERR("Resolution or Format supported!");
```
Resolution or format not supported

mjs513 replied:
actually just Resolution now supported - fixed

avolmat-st commented on drivers/video/ov7670.c:
```diff
@@ -395,69 +378,53 @@ static int ov7670_set_fmt(const struct device *dev, struct video_format *fmt)
 
-	LOG_ERR("Unsupported format");
-	return -ENOTSUP;
+	return ret;
```
If there is a check as done above to display an error and return ret, then this last return ret isn't necessary, it could be 0.
An alternative is to simply do the LOG_ERR within the previous if statement and not return and leave this last return ret.

mjs513 replied:
Made it ret 0.  Think more straightforward.

avolmat-st commented on drivers/video/ov7670.c:
```diff
@@ -617,26 +581,36 @@ static DEVICE_API(video, ov7670_api) = {
+			      POST_KERNEL, CONFIG_VIDEO_INIT_PRIORITY,		\
+			      &ov7670_api);         \
+                                   \
+	VIDEO_DEVICE_DEFINE(ov7670_##n, DEVICE_DT_INST_GET(n), NULL); \
```
It seems to me that this section only change inst into n and change alignment. This is not necessary or at least out of the scope of switching to the cci api.

avolmat-st reviewed (changes requested):
Hi @mjs513,
thanks for the update. Functionally this looks good to me. However some modifications are introduced within this commit which doesn't have to be I think.
This commit is all about switching to the CCI interface for talking to the sensor hence modification should stick to this.

mjs513 commented on drivers/video/ov7670.c:
```diff
@@ -190,32 +127,21 @@ const struct ov7670_resolution_cfg OV7670_RESOLUTION_VGA = {
 
 /* OV7670 definitions */
-#define OV7670_PROD_ID 0x76
+#define OV7670_PROD_ID		0x76
```
Ok put it back to 1 tab.  was trying to get it to align with OV7670_MVFP_HFLIP

mjs513 commented on drivers/video/ov7670.c:
```diff
@@ -190,32 +127,21 @@ const struct ov7670_resolution_cfg OV7670_RESOLUTION_VGA = {
 #define OV7670_MVFP_VFLIP 0x10
 
-#define OV7670_VIDEO_FORMAT_CAP(width, height, format)                                             \
+#define OV767X_VIDEO_FORMAT_CAP(width, height, format)                                             \
```
Ok fixed - also fixed the fmt table so it now follows suit.

mjs513 commented on drivers/video/ov7670.c:
```diff
@@ -379,9 +363,8 @@ static int ov7670_set_fmt(const struct device *dev, struct video_format *fmt)
-	const struct ov7670_resolution_cfg *resolution;
 	int ret;
-	uint8_t i = 0U;
+	uint8_t icount = 0U;
```
Changed it just to be clearer but will revert the changes.

avolmat-st commented on drivers/video/ov7670.c:
```diff
@@ -4,25 +4,18 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#define DT_DRV_COMPAT ovti_ov7670
```
This is related to the introduction of the ov7675, not the move to cci api

avolmat-st commented on drivers/video/ov7670.c:
```diff
@@ -628,15 +588,25 @@ static DEVICE_API(video, ov7670_api) = {
+	VIDEO_DEVICE_DEFINE(ov7670_##inst, DEVICE_DT_INST_GET(inst), NULL); \
+
+
+#undef DT_DRV_COMPAT
```
this line and the one below are not related to the cci api usage.

avolmat-st commented on drivers/video/ov7670.c:
```diff
@@ -369,6 +295,63 @@ static const struct ov7670_reg ov7670_init_regtbl[] = {
+	{OV7670_SCALING_PCLK_DELAY, 0x02},
+};
+
+static const struct video_format_cap fmts[] = {
```
Is there any reason why having this structure moved at a different place within the file ?

avolmat-st commented on drivers/video/ov7670.c:
```diff
@@ -628,15 +588,25 @@ static DEVICE_API(video, ov7670_api) = {
-                                                                                                   \
-	VIDEO_DEVICE_DEFINE(ov7670_##inst, DEVICE_DT_INST_GET(inst), NULL);
-
+#define OV7670_INIT(inst)                                \
```
Is there any reason why changing this code ?

avolmat-st reviewed (changes requested):
Hi @mjs513,
sorry I may sound very annoying with my reviews, however as I said previously, commit should be focused on one kind of modifications (this is also explained in the Zephyr contribution page but is also a general good thing when dealing with commits).
Here for example, the OV7670_INIT maybe sounds like actually just changing the indentation but turned out that there were a change that I didn't noticed at first:

struct ov7670_data ov7670_data_##inst

got replaced by

static struct ov7670_data ov7670_data_##inst

avolmat-st commented on drivers/video/ov7670.c:
```diff
@@ -396,68 +380,49 @@ static int ov7670_set_fmt(const struct device *dev, struct video_format *fmt)
 	memcpy(&data->fmt, fmt, sizeof(data->fmt));
 
 	/* Set output resolution */
+	ret = -ENOTSUP;
```
Seems that this should be set to -ENOTSUP above at line 367

```c
int ret = -ENOTSUP;
```

mjs513 replied:
ok

avolmat-st commented on drivers/video/ov7670.c:
```diff
@@ -396,68 +380,49 @@ static int ov7670_set_fmt(const struct device *dev, struct video_format *fmt)
 
 	/* Set output resolution */
+	ret = -ENOTSUP;
+	i = 0;
```
Apparently i is already initialized with 0 at L368 hence this line is not needed.

mjs513 replied:
deleted

avolmat-st commented on drivers/video/ov7670.c:
```diff
@@ -396,68 +380,49 @@ static int ov7670_set_fmt(const struct device *dev, struct video_format *fmt)
-	LOG_ERR("Unsupported format");
-	return -ENOTSUP;
+	if (ret < 0) {
+		LOG_ERR("Resolution supported!");
```
I think the error should be reporting that the Resolution is **not** supported here.

avolmat-st commented on drivers/video/ov7670.c:
```diff
@@ -396,68 +380,49 @@ static int ov7670_set_fmt(const struct device *dev, struct video_format *fmt)
-	return -ENOTSUP;
+	if (ret < 0) {
+		LOG_ERR("Resolution supported!");
+	return ret;
```
Indentation fix.

mjs513 replied:
fixed.  ran it though checkpatch but probably should have done a clang-format

avolmat-st commented on drivers/video/ov7670.c:
```diff
@@ -628,15 +590,22 @@ static DEVICE_API(video, ov7670_api) = {
-	DEVICE_DT_INST_DEFINE(inst, ov7670_init, NULL, &ov7670_data_##inst, &ov7670_config_##inst, \
-			      POST_KERNEL, CONFIG_VIDEO_INIT_PRIORITY, &ov7670_api);               \
-                                                                                                   \
+#define OV7670_INIT(inst)                                \
```
As I said before, I think there is no reason to update this macro in this patch. This commit is about moving to the CCI read/write api. Here this modification is simply changing the indentation of the macro. Can you please avoid doing that ?

My understanding is that while adding OV7675 support (in a further commit), you will have to update this macro, so you can do this (fixing the indentation) in that commit rather than this one. Doing it here makes think that there is a modification in this part of the code.

mjs513 replied:
Ok.  Just did it here since it was easier to read for the ov7670.  But reverted back to the original

avolmat-st reviewed (changes requested):
Can you also update the commit message. You can simplify it by mentioning the move to the cci helpers ?
Thanks a lot.

mjs513 commented:
@avolmat-st 
What would you like to see in the next commit - was thinking
commit 1.  Change things like ov7670_config to ov767x_config without adding in the ov7675 stuff.  This would include moding the define slightly.  No change to using #define OV767X_INIT yet
commit 2.  Add in #if DT_HAS_COMPAT_STATUS_OKAY(ovti_ov7670) where necessary in prep
commit 3.  Make changes to the defihe:  #define OV767X_INIT(inst, id)   to use id.  
commit 4.  Final commit add in the OV7675 and necessary changes to rest of driver for using the ov7675.

josuah commented:
> What would you like to see in the next commit - was thinking

1. Code reformatting changes usually take their own commit, unless these are just 1 very small change nearby a line modified
2. Here there was also a conversion for using the CCI function family (thank you!) which is unrelated to the conversion of the driver (even though it helps), and could take their own commit
3. Then there is the transformation of the driver happening, which can be a single commit, or if you wish, segment it further in smaller steps, and the sequence you proposed sounds reasonable. It is also possible to have just 2 commits for this step: (3.) make the driver support multiple similar sensors, (4.) add the extra OV7675 sensor using the newly introduced setup.

This will be the first Zephyr driver to support multiple sensors, so the time spent on isolating a commit that add another sensor to an existing driver will help the next users, so it is a bit like writing a step-by-step tutorial on how to do it where every step is a commit in a way.

Thanks all for your patience!

avolmat-st commented on drivers/video/ov767x.c:
```diff
@@ -415,6 +509,40 @@ static int ov767x_set_fmt(const struct device *dev, struct video_format *fmt)
 	}
+#endif
+
+#if DT_HAS_COMPAT_STATUS_OKAY(ovti_ov7675)
```
This won't work since if there are both sensors available on a platform, then both if preproc statements will be true, leading to first setting register based on the OV7670 table, then again, based on the OV7675 table.

josuah reviewed (commented):
Thank you for taking the time for this extra refactoring. It looks like we are almost ready with just a few spacing changes added/removed by mistake, but other than that I do not find any more issue.

avolmat-st reviewed (changes requested):
Thanks for the update. Few cleanup and preproc statements to correct / add to avoid having a big driver when not necessary but apart from that it seems good to me.

mjs513 commented:
@avolmat-st 

Question - not sure about the license in the driver.  At this point think the only thing that applies to nxp is the reg settings for the ov7670.  for the OV7675 they came from the arduino camera library?  Any suggestion on best way to address that - put both in?

avolmat-st commented on drivers/video/ov7670.c:
```diff
@@ -568,17 +530,17 @@ static int ov7670_init(const struct device *dev)
 	k_msleep(5);
 
 	ret = ov7670_set_fmt(dev, &fmt);
+
```
Please avoid adding empty line here.

avolmat-st commented on drivers/video/ov767x.c:
```diff
@@ -386,6 +477,23 @@ static int ov767x_set_fmt(const struct device *dev, struct video_format *fmt)
+		ret = video_write_cci_multiregs8(&config->bus, ov767x_yuv422_regs,
+						 ARRAY_SIZE(ov767x_yuv422_regs));
+	} else {
+		LOG_ERR("Image formaat not supported");
```
typo: formaat -> format

avolmat-st commented on drivers/video/ov767x.c:
```diff
@@ -132,10 +132,10 @@ struct ov767x_data {
-#define OV7675_RGB444               0x8C /* REG444 */
-#define OV7675_COM3_DCW_EN          0x04 /* DCW enable */
+/* Addition defines to support OV7675 */
+#define OV7670_RGB444               0x8C /* REG444 */
```
This change belong to the previous commit which add this register

avolmat-st commented on drivers/video/ov767x.c:
```diff
@@ -192,7 +192,7 @@ static const struct video_reg8 ov767x_init_regtbl[] = {
 	/* configure the output timing */
 	/* PCLK does not toggle during horizontal blank, one PCLK, one pixel */
-	{OV7670_COM10, 0x20}, /* COM10 */
+	{OV7670_COM10, 0x03}, /* COM10 */
```
Not to be done here as previously discussed.

mjs513 replied:
Thought I had put this back.

avolmat-st commented on drivers/video/ov767x.c:
```diff
@@ -458,42 +468,21 @@ static int ov767x_get_caps(const struct device *dev, struct video_caps *caps)
 
-static int ov767x_set_fmt(const struct device *dev, struct video_format *fmt)
+#if DT_HAS_COMPAT_STATUS_OKAY(ovti_ov7670)
+static int ov7670_set_fmt(const struct device *dev, struct video_format *fmt)
```
Isn't it better to do this change prior to adding the ov7675 code ? Basically, within when changing overall from ov7670 to ov767x. Moreover currently some parts are duplicated. memcmp / memcpy of fmt is done in all 3 functions for example. Could you rework the split between functions ?

mjs513 replied:
Note to self - done in resturecture commit

avolmat-st commented on drivers/video/ov767x.c:
```diff
@@ -192,7 +192,7 @@ static const struct video_reg8 ov767x_init_regtbl[] = {
 	/* configure the output timing */
 	/* PCLK does not toggle during horizontal blank, one PCLK, one pixel */
-	{OV7670_COM10, 0x03}, /* COM10 */
+	{OV7670_COM10, 0x20}, /* COM10 */
```
To be fixed.

mjs513 replied:
fixed

avolmat-st commented on drivers/video/ov767x.c:
```diff
@@ -476,13 +476,35 @@ static int ov7670_set_fmt(const struct device *dev, struct video_format *fmt)
 	int ret = -ENOTSUP;
 	uint8_t i = 0U;
 
+	if (fmt->pixelformat != VIDEO_PIX_FMT_RGB565 && fmt->pixelformat != VIDEO_PIX_FMT_YUYV) {
```
Unrelated to what is stated in the commit message. Actually could you clarify this commit, I am having difficulties to understand what has changed and why.

avolmat-st commented:
> @avolmat-st
> 
> Question - not sure about the license in the driver. At this point think the only thing that applies to nxp is the reg settings for the ov7670. for the OV7675 they came from the arduino camera library? Any suggestion on best way to address that - put both in?

Since you mention NXP, I am not sure if you are talking about licence or copyright. Could you clarify ?
As far as copyright is concerned, you can add your copyright below the one from NXP.
Regarding licence, what is the licence of that arduino camera library from which ov7675 settings are coming from ?

mjs513 commented on drivers/video/ov767x.c:
```diff
@@ -132,10 +132,10 @@ struct ov767x_data {
+/* Addition defines to support OV7675 */
+#define OV7670_RGB444               0x8C /* REG444 */
 #define OV7670_COM1                 0x04 /*Com Cntrl1 */
+#define OV7675_COM3_DCW_EN          0x04 /* DCW enable */
```
Think this was done in a later commit

josuah commented on drivers/video/CMakeLists.txt:
If there is a (rare) setup with both `CONFIG_VIDEO_OV7670` and `CONFIG_VIDEO_OV7675`, then the driver would be compiled twice, and possibly build error?

The solution for it: define a `Kconfig.ov767x` that lists `CONFIG_VIDEO_OV767X` but also `CONFIG_VIDEO_OV7670` and `CONFIG_VIDEO_OV7675`.

Example for displays:
https://github.com/zephyrproject-rtos/zephyr/blob/fa61e87f1f2b28e7460c90613ed5518dc19623a4/drivers/display/Kconfig.ili9xxx#L8-L56

avolmat-st replied:
@mjs513 Can you also tackle this comment from @josuah as well ?

avolmat-st replied:
Can you add this in your PR then ? (I am not sure to understand the **Single CMakefile.txt file with 2 entries**).

Btw, an entry into the migration-guide will be needed since this change the existing OV7670 into OV767X, however since we are now in the freeze period prior to the v4.3.0 release, this will have to put on a side until the release of the v4.3.0. Only once v4.3.0 get released the doc/release/migration-guide-4.4 will be created and you will be able to rebase and add the entry into it. Then PR will be able to get merged.

josuah commented on drivers/video/ov767x.c:
Just to be sure, is this the same product ID for both variants? Otherwise, this needs an `if (config->camera_id)` wrapper.

josuah reviewed (commented):
It seems like some commits were added on top, which means the commit history has "add X" then "remove X and add Y instead", which can be combined into a single "add Y".

This is the ["no fixup commit"](https://docs.zephyrproject.org/latest/contribute/contributor_expectations.html#pr-requirements) recommendation.

In case this made it easier, in the meantime that you get more familiar with editing git history (can be tedious when refactoring a lot of things), how about only pushing the first commit (fixing coding style), letting us review everything about it, then pushing the second commit, and wait for review to complete.

This way you would be able to merge all WIP commits, and not have to edit a previous commit in the history.

mjs513 commented on drivers/video/ov767x.c:
The reg settings I grabbed from Arduino_camera library but I believe came from OpenMv configuration for the ov7675.  I could do a double check and verify settings are not different and then I could use a MIT copyright.   Also, looking at their driver I think it might be a better way to go.  But first would like to get all the obvious issues resolved like you and @avolmat-st are finding.  Then do a squash of the all the 7675 commits for clarity before revamping the reg settings for both cameras?  Thoughts.

Thanks for all your help.

avolmat-st commented:
Since you plan to rework the serie, I will review it again after the rework.

avolmat-st commented on drivers/video/ov7670.c:
```diff
@@ -372,47 +382,47 @@ static int ov7670_set_fmt(const struct device *dev, struct video_format *fmt)
-			if (ret < 0) {
-				LOG_ERR("Resolution not set!");
-				return ret;
+	if (config->camera_model == OV767X_MODEL_OV7670) {
```
No need to check for camera_model here since this function is dedicated to the OV7670.

avolmat-st commented on drivers/video/ov767x.c:
```diff
@@ -425,6 +515,43 @@ static int ov7670_set_fmt(const struct device *dev, struct video_format *fmt)
 		}
 	}
 
+	if (config->camera_model == OV767X_MODEL_OV7675) {
```
It seems that this is added within the ov7670_set_fmt function right ? Considering that there is already a ov767x_set_fmt (common entry point) and a ov7670_set_fmt (ov7670 specific function), shouldn't this added into a new ov7675_set_fmt function ?

mjs513 commented on drivers/video/ov767x.c:
```diff
@@ -316,6 +342,37 @@ static const struct video_reg8 ov767x_init_regtbl[] = {
 	{0xb8, 0x0a},
 };
 
+static const struct video_reg8 ov767x_rgb565_regs[] = {
```
Think something didn;t get updated correctly..... as I had rearranged everything....... for the restructure.  will have to redo this commit.

mjs513 replied:
fixed

mjs513 commented on drivers/video/ov767x.c:
```diff
@@ -1,7 +1,5 @@
-Copyright 2021 Arduino SA/*
+/*
```
Breaking out of the functions should be done in the Restructure commit as I believe it is.

mjs513 commented on drivers/video/ov767x.c:
```diff
@@ -316,6 +342,37 @@ static const struct video_reg8 ov767x_init_regtbl[] = {
+};
+
 /* Resolution settings for camera, based on those present in MCUX SDK */
 static const struct video_reg8 ov7670_regs_qcif[] = {
```
fixed

avolmat-st commented on drivers/video/ov767x.c:
```diff
@@ -1,4 +1,4 @@
-/*
+Copyright 2021 Arduino SA/*
```
You don't need to add it again since it is already there at line 3.

avolmat-st commented on drivers/video/ov7670.c:
```diff
@@ -372,20 +382,13 @@ static int ov7670_set_fmt(const struct device *dev, struct video_format *fmt)
 		return -ENOTSUP;
 	}
 
-	if (!memcmp(&data->fmt, fmt, sizeof(data->fmt))) {
```
Seems this piece of code memcmp, and then memcpy got log. This seems to be non sensor specific so could be done at the beginning of the ov767x_set_fmt function.

avolmat-st commented on drivers/video/ov767x.c:
```diff
@@ -167,7 +193,7 @@ static const struct video_reg8 ov767x_init_regtbl[] = {
 	/* configure the output timing */
 	/* PCLK does not toggle during horizontal blank, one PCLK, one pixel */
-	{OV7670_COM10, 0x20}, /* COM10 */
+	{OV7670_COM10, 0x03}, /* COM10 */
```
No. Not to be done here.

avolmat-st commented:
In addition to comment above, I don't understand why there is a 5th commit. I mean, commit 2 is already doing the ov767x_set_fmt so, why putting the ov7675 but into the ov767x_set_fmt in 4th commit and then moving them to a different function in commit 5 ? It seems that commit 5 is changing stuff that have just been added in commit 4.

mjs513 commented:
> Ok, in that case I think it is better to have them squashed since it is all about introducing the OV7675 and this avoid having one thing done in commit 4 then reworked in commit 5.
> Thanks,

Done!

avolmat-st commented on drivers/video/ov767x.c:
```diff
@@ -1,6 +1,8 @@
  * Copyright 2024 NXP
  * Copyright 2021 Arduino SA
+ * Copyright (c) 2013-2021 Ibrahim Abdelkader <iabdalkader@openmv.io>
+ * Copyright (c) 2013-2021 Kwabena W. Agyeman <kwagyeman@openmv.io>
```
Are those two copyright lines (and maybe Arduino SA as well ?) here related to the settings for the OV7675 ?

mjs513 replied:
Yes the are.  Also the RGB setting for the 7670 came from openmv as well.

avolmat-st commented on drivers/video/ov767x.c:
```diff
@@ -377,9 +489,26 @@ static int ov7670_set_fmt(const struct device *dev, struct video_format *fmt)
-	if (fmt->pixelformat != VIDEO_PIX_FMT_RGB565 && fmt->pixelformat != VIDEO_PIX_FMT_YUYV) {
-		LOG_ERR("Only RGB565 and YUYV supported!");
-		return -ENOTSUP;
+	if (!memcmp(&data->fmt, fmt, sizeof(data->fmt))) {
```
Can't this memcmp be part of the ov767x_set_fmt instead ?

mjs513 replied:
Actually yes.  Will update

avolmat-st reviewed (changes requested):
Few small points

avolmat-st commented:
> > Haven't reviewed everything, however it seems to me that the memcpy on set_fmt is still missing. (already mentioned [#96319 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/96319#discussion_r2445176981) and [#96319 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/96319#discussion_r2445455060))
> 
> Was sure that the extra memcpy was needed. Anyways added it back in:
> 
> ```
> 	if (!memcmp(&data->fmt, fmt, sizeof(data->fmt))) {
> 		/* nothing to do */
> 		return 0;
> 	}
> 
> 	memcpy(&data->fmt, fmt, sizeof(data->fmt));
> ```
> 
> This only appears in OV767x_set_fmt in commit Add OV7675 changes to OV767x.
> 
> Ok just checked looks like I need to add it to as well for completeness: Update OV7670 to use generic ov767x functions
> 
> EDIT: All updated

This memcpy was part of the original code and without it data->fmt is never set so yes it is mandatory.

avolmat-st reviewed (commented):
Few small / minor points left, but apart from that (and Copyright point to be discussed), looks ok.

avolmat-st reviewed (changes requested):
Could you also simplify the commit messages ? It is not necessary to mention what has been discussed or steps to achieve this. The commit message should give information about what is done in this commit.

mjs513 commented on drivers/video/ov767x.c:
```diff
@@ -167,12 +195,12 @@ static const struct video_reg8 ov767x_init_regtbl[] = {
 	/* configure the output timing */
 	/* PCLK does not toggle during horizontal blank, one PCLK, one pixel */
-	{OV7670_COM10, 0x20}, /* COM10 */
+	{OV7670_COM10, 0x03}, /* COM10 */
```
Could have sworn I fixed it before pushing.  Sorry

mjs513 commented on drivers/video/ov767x.c:
```diff
@@ -424,9 +612,9 @@ static int ov7670_set_fmt(const struct device *dev, struct video_format *fmt)
+	const struct ov767x_config __maybe_unused *config = dev->config;
 	struct ov767x_data *data = dev->data;
-	int ret;
+	[[maybe_unused]] int ret;
```
Ok.  Was giving an error but found if I insert a blank like between struct and int ret __maybe_unused checkpatch does not complain.  So will push that change

mjs513 commented:
> Could you also simplify the commit messages ? It is not necessary to mention what has been discussed or steps to achieve this. The commit message should give information about what is done in this commit.

Was trying to keep a running tally of what changes were being made and would simplify at the end.

josuah commented:
Hello @mjs513. It looks like you needed to import the latest `main` branch into your work. This is good idea but you will need to do this with a `git rebase` rather than a `git merge`. This is to avoid the "merge commit" as seen at the end:

![screenshot_2025_1026_1001_35](https://github.com/user-attachments/assets/4c8f4d14-b204-4c06-ae1a-5fb7db180c57)

Could you try the operation again? You might need a `git reset --hard <commit_before_merge>` to cancel the "git merge", then try again with `git rebase origin/main` to use "rebase" instead of "merge".

mjs513 commented:
> Hello @mjs513. It looks like you needed to import the latest `main` branch into your work. This is good idea but you will need to do this with a `git rebase` rather than a `git merge`. This is to avoid the "merge commit" as seen at the end:
> 

> Could you try the operation again? You might need a `git reset --hard <commit_before_merge>` to cancel the "git merge", then try again with `git rebase origin/main` to use "rebase" instead of "merge".

Thats weird didnt notice that will double check\

avolmat-st commented on drivers/video/ov7670.c:
```diff
@@ -1,5 +1,7 @@
 /*
  * Copyright 2024 NXP
+ * Copyright 2021 Arduino SA
```
At least, in this commit, aka commit number 2, I think this is only NXP + your copyright. There is nothing coming from OpenMv / Arduino at this point I think.

avolmat-st replied:
@mjs513, is the Copyright Arduino SA relevant here ?

mjs513 replied:
Changes made per another comment

mjs513 commented:
@josuah - @avolmat-st 

Have some questions on fixing the identified twister and documentation errors.

think the twister errors may be related to me specifing the reset and pwrdn pins for 0v7675 while for the 0v7670 they are not specified.

As for the documentation errors - no clue how to fix.

Suggestions?

josuah commented:
It seems like the latest commit adds not-yet-merged drivers and deletes existing drivers:
https://github.com/zephyrproject-rtos/zephyr/pull/96319/commits/962b6b1def3ec762490fdf9b33c609eb401936bc

Given the twister errors and docs error both mention Kconfig, I suggest to check that the commit only modifies what you wished to modify, and try again.

For this, you could copy-paste the CMakeLists.txt and Kconfig from current `main` and then insert the needed changes again.

mjs513 commented:
Ok question on the app.overlay
right now I have for both the ov7670 (already there) and the OV7675:
```
			test_i2c_ov7675: ov7675@21 {
				compatible = "ovti,ov7675";
				reg = <0x21>;
			};
```
do I need to add:
```
				port {
					hm01b0_ep_out: endpoint {
						remote-endpoint-label = "";
					};
				};
```
Really not sure on this one since twister looks like it failing for certain boards.  Or should I worry about it.

josuah commented on tests/drivers/build_all/video/app.overlay:
```diff
@@ -148,8 +148,6 @@
 				compatible = "ovti,ov7675";
 				reg = <0x21>;
-				reset-gpios = <&test_gpio 0 0>;
-				pwdn-gpios = <&test_gpio 1 0>;
```
Doesn't it work anymore? If it still works you can keep these lines to test extra code.

mjs513 commented:
@josuah etc
Made a change to app.overlay after runnting twister locally but twister still failing but not sure if me or something else.  Am listing the build log for one of the boards - all other boards failing have same issue:
```
Loading Zephyr default modules (Zephyr base).
-- Application: /home/my_new_zephyr_folder/zephyr/tests/drivers/build_all/video
-- CMake version: 3.28.3
-- Found Python3: /root/.local/share/pipx/venvs/west/bin/python (found suitable version "3.12.3", minimum required is "3.10") found components: Interpreter 
-- Cache files will be written to: /root/.cache/zephyr
-- Zephyr version: 4.2.0 (/home/my_new_zephyr_folder/zephyr)
-- Found west (found suitable version "1.3.0", minimum required is "0.14.0")
-- Board: arduino_nicla_vision, qualifiers: stm32h747xx/m7
-- Found host-tools: zephyr 0.17.0 (/home/zephyr-sdk-0.17.0)
-- Found toolchain: zephyr 0.17.0 (/home/zephyr-sdk-0.17.0)
-- Found Dtc: /home/zephyr-sdk-0.17.0/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6") 
-- Found BOARD.dts: /home/my_new_zephyr_folder/zephyr/boards/arduino/nicla_vision/arduino_nicla_vision_stm32h747xx_m7.dts
-- Found devicetree overlay: /home/my_new_zephyr_folder/zephyr/tests/drivers/build_all/video/app.overlay
-- Generated zephyr.dts: /home/my_new_zephyr_folder/zephyr/twister-out/arduino_nicla_vision_stm32h747xx_m7/zephyr/tests/drivers/build_all/video/drivers.video.stm32_dcmi.build/zephyr/zephyr.dts
-- Generated pickled edt: /home/my_new_zephyr_folder/zephyr/twister-out/arduino_nicla_vision_stm32h747xx_m7/zephyr/tests/drivers/build_all/video/drivers.video.stm32_dcmi.build/zephyr/edt.pickle
-- Generated devicetree_generated.h: /home/my_new_zephyr_folder/zephyr/twister-out/arduino_nicla_vision_stm32h747xx_m7/zephyr/tests/drivers/build_all/video/drivers.video.stm32_dcmi.build/zephyr/include/generated/zephyr/devicetree_generated.h
/home/my_new_zephyr_folder/zephyr/twister-out/arduino_nicla_vision_stm32h747xx_m7/zephyr/tests/drivers/build_all/video/drivers.video.stm32_dcmi.build/zephyr/zephyr.dts:2253.31-2287.6: Warning (unique_unit_address_if_enabled): /test/i2c@10002000/bridge@8: duplicate unit-address (also used in node /test/i2c@10002000/ov9655@8)
Parsing /home/my_new_zephyr_folder/zephyr/Kconfig
/home/my_new_zephyr_folder/zephyr/scripts/kconfig/kconfig.py: drivers/video/Kconfig:85: 'drivers/video/Kconfig.stm32_venc' not found (in 'source "drivers/video/Kconfig.stm32_venc"'). Check that environment variables are set correctly (e.g. $srctree, which is set to '/home/my_new_zephyr_folder/zephyr'). Also note that unset environment variables expand to the empty string.
CMake Error at /home/my_new_zephyr_folder/zephyr/cmake/modules/kconfig.cmake:389 (message):
  command failed with return code: 1
Call Stack (most recent call first):
  /home/my_new_zephyr_folder/zephyr/cmake/modules/zephyr_default.cmake:131 (include)
  /home/my_new_zephyr_folder/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:66 (include)
  /home/my_new_zephyr_folder/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:92 (include_boilerplate)
  CMakeLists.txt:4 (find_package)


-- Configuring incomplete, errors occurred!
```

josuah commented:
It looks like CI became extra pedantic about devicetree syntax.

```
 Not correctly formatted.:Index: a/tests/drivers/build_all/video/app.overlay
===================================================================
--- a/tests/drivers/build_all/video/app.overlay
+++ a/tests/drivers/build_all/video/app.overlay
@@ -137,8 +137,9 @@
 				compatible = "himax,hm01b0";
 				reg = <0x24>;
 				status = "okay";
 				data-bits = <0x4>;
+
 				port {
 					hm01b0_ep_out: endpoint {
 						remote-endpoint-label = "";
 					};
@@ -172,5 +173,5 @@
 				};
 			};
 		};
 	};
-};
\ No newline at end of file
+};
```

josuah commented:
Regarding VENC, is it possible that you are not having `venc` in your version?

You might then do a `git fetch upstream` then `git rebase upstream/main` to make sure to work on the latest sources.

avolmat-st commented on drivers/video/CMakeLists.txt:
```diff
@@ -19,6 +19,7 @@ zephyr_library_sources_ifdef(CONFIG_VIDEO_STM32_VENC	video_stm32_venc.c)
 zephyr_library_sources_ifdef(CONFIG_VIDEO_OV5640	ov5640.c)
 zephyr_library_sources_ifdef(CONFIG_VIDEO_OV7670	ov767x.c)
 zephyr_library_sources_ifdef(CONFIG_VIDEO_OV9655	ov9655.c)
+zephyr_library_sources_ifdef(CONFIG_VIDEO_OV7675	ov767x.c)
```
The modification (of your last commit) should be done within this commit to avoid introducing this unlikely behavior of having the driver build twice if both OV7670 & OV7675 are both enabled.
So, last commit of the PR should be squashed in this commit "drivers: video Add OV7675 changes to OV767x"

avolmat-st replied:
This is still done in a separate commit apparently.

mjs513 replied:
fixed
<img width="657" height="69" alt="image" src="https://github.com/user-attachments/assets/69bbf212-360e-41db-a85b-4c6851a8890c" />

avolmat-st reviewed (changes requested):
Thanks for the update. Few more points reported above.

As well as:

- since in Copyright statements you use "Michael Smorto <CyberMerln@gmail.com>", why not having the same full name in all your Signed-off-by instead of "Mike S <CyberMerln@gmail.com>".  cf https://docs.zephyrproject.org/latest/contribute/guidelines.html#dco-sign-off

- The PR changes the Zephyr configuration, CONFIG_VIDEO_OV7670 is now gone and replaced by CONFIG_VIDEO_OV767X, so this should be reported in the doc/releases/migration-guide-4.4.rst file in order to help users to migrate from v4.3.0 to v4.4.0. Please also add an entry into the file. You can take a look at other migration files for examples.

mjs513 commented on drivers/video/ov767x.c:
```diff
@@ -1,6 +1,7 @@
 /*
  * Copyright 2024 NXP
  * Copyright 2021 Arduino SA
+ * Copyright (c) 2017 OpenMV
```
ok next update

mjs513 committed:
drivers: video: Add OV7675 changes to OV767x

Adds support for the ov7675 camers.

Signed-off-by: Michael Smorto <CyberMerln@gmail.com>

mjs513 committed:
drivers: video: Changes the driver name to ov767x.c

Changes the driver name to ov767x.

Signed-off-by: Michael Smorto <CyberMerln@gmail.com>

mjs513 committed:
drivers: video: Update OV7670 to use generic ov767x functions

Minimal changes to OV760 drive to use generic ov767x structures.
- This includes changes to function names.
-  Includes changes to defines as necessary.
-  Deleted camera_id as no longer required

Signed-off-by: Michael Smorto <CyberMerln@gmail.com>

mjs513 committed:
drivers: video: Restructure OV760.c driver before adding OV7675 support

Modifications for use of video cci helpers in video_common.h.

Signed-off-by: Michael Smorto <CyberMerln@gmail.com>

avolmat-st reviewed (changes requested):
Some comments not yet fixed, can you please fix them before requesting a new review ?
At least from what I remember, there are still the following points not fixed, commented.

https://github.com/zephyrproject-rtos/zephyr/pull/96319#discussion_r2493364489

https://github.com/zephyrproject-rtos/zephyr/pull/96319#pullrequestreview-3420418311

https://github.com/zephyrproject-rtos/zephyr/pull/96319/commits/1b9cd7861358b7693c066a8bc9da261623128199#r2493351602

mjs513 commented:
@avolmat-st 
> Some comments not yet fixed, can you please fix them before requesting a new review ? At least from what I remember, there are still the following points not fixed, commented.
> 
> [#96319 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/96319#discussion_r2493364489)
> 
> [#96319 (review)](https://github.com/zephyrproject-rtos/zephyr/pull/96319#pullrequestreview-3420418311)
> 
> [1b9cd78#r2493351602](https://github.com/zephyrproject-rtos/zephyr/commit/1b9cd7861358b7693c066a8bc9da261623128199#r2493351602)

In regards to #96319:
1.  Copyright notice was fixed to your specification in that last commit.
2.  I do not doc/releases/migration-guide-4.4.rst in version 4.3.0rc1 which is what I am basedlined.  Currently in this repo is blank - so you still want to add I can create

In regards to the last link.  Not sure what the issue.  Copyright notice was modified.

mjs513 commented:
@josuah - @avolmat-st 
Rebased with latest changes to fix merge commit and able to edit rst now.

mjs513 commented:
@avolmat-st 
> Indeed, link doesn't work. The point regarding Copyright was whether or not Copyright 2021 Arduino SA is really necessary to add, especially from the 2nd commit into which you are switching structures / macro naming, hence not related to Arduino SA I believe. Is that correct ?

> Regarding migration-guide-4.4.rst, if it isn't yet available, then it can be done later on. This PR will have to wait before being merged until the release of v4.3.0 which will happen next week.

I did a rebase so was able to incorporated the change for  migration-guide-4.4.rst in the last push.

I updated the copyright notices so that NXP and myself are at the top and added comments for RGB/YUV settings from openmv and ov7675 resolutions from arduino.  Will be in the next push

mjs513 commented:
@avolmat-st 

Just clicked on re-review - cant remember if I did earlier - sorry if its a dup request

avolmat-st commented:
Thanks for the updates. One of my comments was about the fact that Arduino is written as copyright in your 2nd commit. I can now well see that you've added a comment within the code but that copyright statement is still present in your 2nd commit at the top of the file, cf this screenshot which shows the modifications performed by your 2nd commit of this PR:

<img width="1786" height="637" alt="image" src="https://github.com/user-attachments/assets/a83282ab-5f62-4048-980b-4906afb5ea1f" />

mjs513 commented:
could have sworn I deleted that - going to double check and if not will push it right up.

Edit:  Ok I only deleted it in the last commit not it the generic function commit - will up but may be later.

avolmat-st reviewed (dismissed):
Thanks, LGTM.

ngphibang commented on drivers/video/ov767x.c:
```diff
@@ -0,0 +1,848 @@
+		return ret;
+	}
+
+	if (pid != OV7670_PROD_ID) {
```
This is common so should it be `OV767X_PROD_ID` ?

ngphibang reviewed (commented):
May I miss something in the comments but did you or someone already test this new change on ov7670 and its supported boards (e.g. mcxn947) to ensure there is no regression (I 'd like to do so but unfortunatly don't have the HW).

Thanks !

ngphibang commented on drivers/video/ov7670.c:
```diff
@@ -631,12 +593,13 @@ static DEVICE_API(video, ov7670_api) = {
 	const struct ov7670_config ov7670_config_##inst = {.bus = I2C_DT_SPEC_INST_GET(inst),      \
 							   OV7670_RESET_GPIO(inst)                 \
-								   OV7670_PWDN_GPIO(inst)};        \
+							   OV7670_PWDN_GPIO(inst)};        \
```
Just in this commit: unrelated changed, indentation is already OK as in the olde code. But the final change is OK, so not blocking.

mjs513 replied:
If alright prefer to leave as is thinkgs are lined up now.

ngphibang reviewed (commented):
Some minors on the commit titles:

drivers: video: Changes ov7670 driver name to ov767x.c
drivers: video: Add OV7675 changes to OV767x

mjs513 commented on drivers/video/ov7670.c:
```diff
@@ -631,12 +593,13 @@ static DEVICE_API(video, ov7670_api) = {
                                                                                                    \
 	VIDEO_DEVICE_DEFINE(ov7670_##inst, DEVICE_DT_INST_GET(inst), NULL);
 
-DT_INST_FOREACH_STATUS_OKAY(OV7670_INIT)
```
Think I did the edit right.

ngphibang reviewed (commented):
Thanks for the update. LGTM, just a minor comment and the question on the common naming of the init register table (cc @avolmat-st )

Should the commit title: `Drivers: video Add OV7675 changes to OV767x`  be `drivers: video: Add OV7675 changes to OV767x` ?

mjs513 committed:
drivers: video: Changed common registers to OV767X

Updated register tables to use use OV767X.

Signed-off-by: Michael Smorto <CyberMerln@gmail.com>

ngphibang reviewed (commented):
Could you resolve the conflict as well ?

mjs513 commented on drivers/video/ov767x.c:
```diff
@@ -0,0 +1,848 @@
+ * This initialization table is based on the MCUX SDK driver for the OV7670.
+ * Note that this table assumes the camera is fed a 6MHz XCLK signal
+ */
+static const struct video_reg8 ov767x_init_regtbl[] = {
```
missed this to mark as resolve

mjs513 commented on drivers/video/ov767x.c:
```diff
@@ -348,78 +348,78 @@ static const struct video_reg8 ov767x_init_regtbl[] = {
-	{OV7670_COM9, 0x6a}, /* 16x gain ceiling; 0x8 is reserved bit */
+	{OV767X_RGB444, 0x00},              /* No RGB444 please */
+	{OV767X_COM1, 0x00},                /* CCIR601 */
+	{OV767X_COM15, OV7675_COM15_OUT_00_FF | OV7675_COM15_FMT_RGB565},
```
Probably will change next for next push - need to retest.

mjs513 commented:
@ngphibang 
Had to do a rebase as well since migration file keeps changing and tests told me I had to rebase
